app/controllers/comments_controller.rb

```diff
@@ -8,10 +10,26 @@ def create
     redirect_to share_path(@share)
   end
 
+  def destroy
+   return render_not_found if @comment.blank?
+   return render_not_found(:forbidden) if @comment.user != current_user
+   @comment.destroy
+   flash[:notice] = "Your comment was deleted."
+   redirect_to :back
+  end
+
   private
 
-  def comment_params
-    params.require(:comment).permit(:body)
-  end
+    def comment_params
+      params.require(:comment).permit(:body)
+    end
+
+    def set_comment
+      @comment = Comment.find(params[:id])
```

Comment:
Noted. I forgot I had to change this on a person project, because the test were producing errors.